src/libraries/Microsoft.PowerFx.Core/Public/Values/DateTimeValue.cs

```diff
@@ -15,6 +15,28 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public class DateTimeValue : PrimitiveValue<DateTime>
     {
+        /// <summary>
+        /// Converts To UTC time if value is not utc and <paramref name="timeZoneInfo"/> is utc.
+        /// Converts from UTC time if value is utc and <paramref name="timeZoneInfo"/> is non utc.
+        /// else returns the value.
+        /// NOTE: if <paramref name="timeZoneInfo"/> is null, Local time zone is used.
+        /// </summary>
+        public DateTime GetConvertedDateTimeValue(TimeZoneInfo timeZoneInfo)
+        {
+            timeZoneInfo ??= TimeZoneInfo.Local;
+
+            if (_value.Kind != DateTimeKind.Utc && timeZoneInfo.BaseUtcOffset == TimeSpan.Zero)
```

Comment:
>timeZoneInfo.BaseUtcOffset == TimeSpan.Zero [](http://example.com/codeflow?start=51&length=43)

do we need this check?  (is it already handled by ConvertTimeToUtc)

Reply:
We don't wanna attempt to convert to UTC if setTimeZone isn't UTC, otherwise ConvertTimeToUtc would throw